### PR TITLE
cli: util clean: Add filter options

### PR DESCRIPTION
**Pull request by tingerrr**
See commit messages.

### CODE REVIEW

**Comment by tingerrr** [comment]
where did my CI go

File: crates/tytanic/src/cli/commands/util/clean.rs [hunk 1]
```diff
@@ -5,22 +5,45 @@ use termcolor::Color;
...
-        len += 1;
+        if args.include_persistent_references && test.kind().is_persistent() {
+            test.delete_reference_document(&project)?;
+            persistent += 1;
+        }
+        temp += 1;
     }
 
     let mut w = ctx.ui.stderr();
     write!(w, "Removed temporary directories for ")?;
```
Comment (Andrew15-5) [inline crates/tytanic/src/cli/commands/util/clean.rs:38]: Does it really need to say that it removed this, even if it didn't remove anything?

**Review by Andrew15-5 — APPROVED** [review]

File: crates/tytanic/src/cli/commands/util/clean.rs (hunk 1 above)
Comment (tingerrr) [reply crates/tytanic/src/cli/commands/util/clean.rs:38]: It could also not write anything, but I figured it should at least give an indication, even if it cleaned nothing at all. That usually means the user used it incorrectly.

Comment (Andrew15-5) [reply crates/tytanic/src/cli/commands/util/clean.rs:38]: Well, in that case, the messages should be something else. It shouldn't lie to me that it cleaned all tests when they are all already clean. So maybe an `.empty()` edge case.

Comment (tingerrr) [reply crates/tytanic/src/cli/commands/util/clean.rs:38]: I'll try to remember that when I rework the fs access stuff in the core lib.

Comment (Andrew15-5) [reply crates/tytanic/src/cli/commands/util/clean.rs:38]: Here you go: https://github.com/tingerrr/tytanic/issues/182
